### ISSUE TITLE
Corruption of DOCX, PPTX and XLSX files

### ISSUE THREAD

**Issue by ECJansen** [codelibs/fess#2111]
When clicking on a search result link for DOCX, PPTX and XLSX files on an SMB server, the file is returned corrupted.  This does not happen for DOC, PPT and XLS files nor any other type of file.  Corruption was verified by comparing the file on the share with the file returned and saved to disc from the search result link.  A file comparison shows the introduction of at least one extra character in the file that is returned.  It happens regardless of the browser, Chrome, Explorer or Firefox.
We are running ElasticSearch 6.3.1 and FESS 12.5.2.
Any ideas why this is happening ?
Thanks,
Eric

**Comment by marevol** [comment]
Fess 12.5.x supports only Elasticsearch 6.6.x.
See [Download](https://fess.codelibs.org/downloads.html) page.

**Comment by ECJansen** [comment]
Aha !  I'll update.  Thanks !

**Comment by ECJansen** [comment]
I deleted the Elasticsearch database and removed Elasticsearch and FESS and installed Elasticsearch 6.6.2 and FESS 12.6.0.  I created a test directory and crawled two test files, a DOC and a DOCX.  When the DOCX file is retrieved and saved, it still is returned corrupted.
Thoughts ?
[Fess-test-document.docx](https://github.com/codelibs/fess/files/3150303/Fess-test-document.docx)
[Fess-test-document-retrieved.docx](https://github.com/codelibs/fess/files/3150304/Fess-test-document-retrieved.docx)

**Comment by marevol** [comment]
Fess 12.6.x supports only Elasticsearch 6.7.x.

Using Fess-test-document.docx, I could not reproduce the problem.
Did you check fess.log?

**Comment by ECJansen** [comment]
Fess.log showed access denied for the thumbnails directory. The owning GID/UID changed for the fess account, probably from the uninstall/install.  The other logs are clean.  I'll try updating ES to 6.7.  The 12.6 installation instructions show ES 6.6 is supported.
Thanks

**Comment by ECJansen** [comment]
I did a wipe and fresh install of both ES 6.7.0, with the listed plug-ins, and FESS 12.6.  I crawled the two test files (.DOC and .DOCX). The .DOC is returned intact.  The .DOCX is modified when returned.  I noticed that when using the "Open with" option to open in Word and receiving the error message, the filename is modified to include a "-<digit>", i.e. "Fess-test-document-1.docx". Every time I test, via clicking on the link, the digit increments by 1.  Is some sort of transformation being performed on the document upon retrieval ?
Thanks
